src/augeas.c

```diff
@@ -2020,6 +2021,35 @@ int aug_escape_name(augeas *aug, const char *in, char **out) {
     return result;
 }
 
+int aug_load_file(struct augeas *aug, const char *file) {
+    int result = -1;
+    struct tree *meta = tree_child_cr(aug->origin, s_augeas);
+    struct tree *load = tree_child_cr(meta, s_load);
+    bool found = false;
+
+    api_entry(aug);
+
+    ERR_NOMEM(load == NULL, aug);
+
+    list_for_each(xfm, load->children)  {
+      if (filter_matches(xfm, file)) {
+        transform_load(aug, xfm, file);
+        found = true;
+        break;
+      } 
+    }
+
+    ERR_THROW(!found, aug, AUG_ESLOADFILE, "can not determine lens to load file %s", file);
+
+    api_exit(aug);
+    result = 0;
+    return result;
```

Comment:
These three lines should just become `result = 0` and fall through to the `error` label so that any cleanup that might be added in the future happens both when we return success and failure.